File: scripts/01-download_data.R
#### Preamble ####
# Purpose: Downloads and saves the data from the Open Data Toronto portal
# Author: Amie Liu
# Date: 16 January 2024
# Contact: amie.liu@mail.utoronto.ca
# License: MIT
# Pre-requisites: None

#### Workspace setup ####
# install.packages("opendatatoronto")
# install.packages("tidyverse")

library(opendatatoronto)
library(tidyverse)

#### Download COVID-19 cases in Toronto data ####
# COVID-19 cases Dataset and code are from: https://open.toronto.ca/dataset/covid-19-cases-in-toronto/
# Get all resources from this package
resources <- list_package_resources("64b54586-6180-4485-83eb-81e8fae3b8fe")

# Identify datastore resources in csv format
datastore_resources <- filter(resources, tolower(format) %in% c('csv', 'geojson'))

# Load the COVID-19 cases.csv
raw_case_data <- filter(datastore_resources, row_number()==2) |> get_resource()

raw_case_data

#### Save data ####
write_csv(raw_case_data, "inputs/data/raw_data.csv")

# Rename the file for raw data
file.rename("inputs/data/raw_data.csv", "inputs/data/raw_case_data.csv")

File: scripts/01-download_data.py
#### Preamble ####
# Purpose: Downloads and saves the data from the Open Data Toronto portal
# Author: Lena Wang
# Date: 16 January 2024
# Contact: amie.liu@mail.utoronto.ca
# License: MIT
# Pre-requisites: None

#### Workspace setup ####
import pandas as pd
from io import StringIO

#### Download COVID-19 cases in Toronto data ####
import requests

base_url = "https://ckan0.cf.opendata.inter.prod-toronto.ca"

url = base_url + "/api/3/action/package_show"
params = { "id": "covid-19-cases-in-toronto"}
package = requests.get(url, params = params).json()

# To get resource data:
for idx, resource in enumerate(package["result"]["resources"]):

       # for datastore_active resources:
       if resource["datastore_active"]:
            # To get all records in CSV format:
            url = base_url + "/datastore/dump/" + resource["id"]
            resource_dump_data = requests.get(url)

            # Writing raw data to CSV
            raw_case_data = pd.read_csv(StringIO(resource_dump_data.text))
            raw_case_data.to_csv("inputs/data/raw_case_data.csv")
            

